ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/deneb/helpers/MiscHelpersDeneb.java

```diff
@@ -65,26 +67,43 @@ private static KZG initKZG(final SpecConfigDeneb config) {
   public boolean isDataAvailable(
       final UInt64 slot,
       final Bytes32 beaconBlockRoot,
-      final List<KZGCommitment> kzgCommitments,
+      final List<KZGCommitment> kzgCommitmentsFromBlock,
       final List<BlobSidecar> blobSidecars) {
-    blobSidecars.forEach(
-        blobSidecar -> {
-          checkArgument(
-              slot.equals(blobSidecar.getSlot()),
-              "Blob sidecar slot %s does not match block slot %s",
-              blobSidecar.getSlot(),
-              slot);
-          checkArgument(
-              beaconBlockRoot.equals(blobSidecar.getBlockRoot()),
-              "Blob sidecar block root %s does not match block root %s",
-              blobSidecar.getBlockRoot(),
-              beaconBlockRoot);
-        });
-    final List<Bytes> blobs =
-        blobSidecars.stream().map(BlobSidecar::getBlob).map(Blob::getBytes).toList();
-    final List<KZGProof> proofs = blobSidecars.stream().map(BlobSidecar::getKZGProof).toList();
-
-    return kzg.verifyBlobKzgProofBatch(blobs, kzgCommitments, proofs);
+    checkArgument(
+        blobSidecars.size() == kzgCommitmentsFromBlock.size(),
+        "Blob sidecars count %s does not match KZG commitments count %s",
+        blobSidecars.size(),
+        kzgCommitmentsFromBlock.size());
+
+    final List<Bytes> blobs = new ArrayList<>();
+    final List<KZGProof> proofs = new ArrayList<>();
+
+    IntStream.range(0, blobSidecars.size())
+        .forEach(
+            index -> {
+              final BlobSidecar blobSidecar = blobSidecars.get(index);
+
+              checkArgument(
+                  slot.equals(blobSidecar.getSlot()),
+                  "Blob sidecar slot %s does not match block slot %s",
+                  blobSidecar.getSlot(),
+                  slot);
+              checkArgument(
```

Comment:
You could also add parent root check

Reply:
I started thinking that these checks could leave in the `BlockBlobSidecarsTracker`. If the tracker guarantees that block and sidecars field are consistent, we can then have this method only dedicated to KZG check, which is the original intention. The good thing on discarding inconsistent blobSidecars at `BlockBlobSidecarsTracker` level is that we could try to refetch them early.

Reply:
thinking more about it, an early refetch might be undesirable here. If the proposer is doing something strange, is better to do nothing, skip the block import, vote on previous head, and then lookup only if next block builds on top of this (which might be not the case most times, unless next proposer is under control of the same malicious\buggy entity)

Reply:
so `ForkChoiceBlobSidecarsAvailabilityChecker::validateBatch` might be a better place to have this checks

Reply:
I decided to move the validation back to miscHelper but on a dedicated method. This simplified the tests in `ForkChoiceBlobSidecarsAvailabilityCheckerTest`